lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -190,11 +201,11 @@ def calculate_percentage_change_in_last_iteration(allvecs):
             args=(forecast_predictor_data, truth_data,
                   forecast_var_data, sqrt_pi, predictor_of_mean_flag),
             method="Nelder-Mead",
-            options={"maxiter": self.MAX_ITERATIONS, "return_all": True})
+            options={"maxiter": self.max_iterations, "return_all": True})
         if not optimised_coeffs.success:
             msg = ("Minimisation did not result in convergence after "
                    "{} iterations. \n{}".format(
-                       self.MAX_ITERATIONS, optimised_coeffs.message))
+                       self.max_iterations, optimised_coeffs.message))
```

Comment:
Sometimes we are getting nans in our initial guess and sometimes not, this can be seen by the intermittent appearance of the full warning shown below before convergence is reached.
```
UserWarning: Minimisation did not result in convergence after 680 iterations. 
Maximum number of iterations has been exceeded.
  warnings.warn(msg)
UserWarning: The final iteration resulted in a percentage change that is greater than the accepted threshold of 5% i.e. [292.69569882  61.02072573   0.0765011    0.02498964   0.00510256   0.00169396]. 
A satisfactory minimisation has not been achieved. 
Last iteration: [-0.00027948  0.00003825  0.00517252 -0.21116614  0.60728499  0.76590817], 
Last-but-one iteration: [ 0.00014504  0.00009813  0.00516856 -0.21111338  0.60731598  0.76589519]
Absolute difference: [0.00042452 0.00005988 0.00000395 0.00005276 0.00003099 0.00001297]
```
For some values of max_iterations we get only the following warning before convergence.
```
UserWarning: Minimisation did not result in convergence after 681 iterations. 
Maximum number of iterations has been exceeded.
```

I don't understand enough of what is going on to guess why some iterations are returning nans and some not.